tests/runtimes/test_application.py

```diff
@@ -0,0 +1,197 @@
+
+
+def test_create_application_runtime():
+    fn: mlrun.runtimes.ApplicationRuntime = mlrun.code_to_function(
+        "application-test", kind="application", image="mlrun/mlrun"
+    )
+    assert fn.kind == mlrun.runtimes.RuntimeKinds.application
+    assert fn.spec.image == "mlrun/mlrun"
+    assert fn.metadata.name == "application-test"
+    # base64 prefix of the reverse proxy code
+    assert (
+        "Ly8gQ29weXJpZ2h0IDIwMjQgSWd1YXppbwovLwovLyBMaWN"
+        in fn.spec.build.functionSourceCode
+    )
+
+    filepath, expected_handler = (
+        mlrun.runtimes.ApplicationRuntime.get_filename_and_handler()
+    )
+    expected_filename = pathlib.Path(filepath).name
+    expected_function_handler = f"{expected_filename.split('.')[0]}:{expected_handler}"
+    assert fn.spec.function_handler == expected_function_handler
+
+
+def test_create_application_runtime_with_command(rundb_mock):
+    fn: mlrun.runtimes.ApplicationRuntime = mlrun.new_function(
+        "application-test", kind="application", image="mlrun/mlrun", command="echo"
+    )
+    fn.deploy()
+    assert fn.spec.config["spec.sidecars"][0]["command"] == "echo"
+    assert fn.kind == mlrun.runtimes.RuntimeKinds.application
+    assert fn.status.application_image == "mlrun/mlrun"
+    assert fn.metadata.name == "application-test"
+    # base64 prefix of the reverse proxy code
+    assert (
+        "Ly8gQ29weXJpZ2h0IDIwMjQgSWd1YXppbwovLwovLyBMaWN"
+        in fn.spec.build.functionSourceCode
+    )
+
+    filepath, expected_handler = (
+        mlrun.runtimes.ApplicationRuntime.get_filename_and_handler()
+    )
+    expected_filename = pathlib.Path(filepath).name
+    expected_function_handler = f"{expected_filename.split('.')[0]}:{expected_handler}"
+    assert fn.spec.function_handler == expected_function_handler
+
+
+def test_deploy_application_runtime(rundb_mock):
+    image = "my/web-app:latest"
+    fn: mlrun.runtimes.ApplicationRuntime = mlrun.code_to_function(
+        "application-test", kind="application", image=image
+    )
+    fn.deploy()
+    assert fn.spec.config["spec.sidecars"] == [
+        {
+            "image": image,
+            "name": "application-test-sidecar",
+            "ports": [{"containerPort": 8080, "name": "http", "protocol": "TCP"}],
+        }
+    ]
+    assert fn.get_env("SIDECAR_PORT") == "8080"
+    assert fn.status.application_image == image
+    assert not fn.spec.image
+
+
+def test_consecutive_deploy_application_runtime(rundb_mock):
+    image = "my/web-app:latest"
+    fn: mlrun.runtimes.ApplicationRuntime = mlrun.code_to_function(
+        "application-test", kind="application", image=image
+    )
+    fn.deploy()
+    assert fn.spec.config["spec.sidecars"] == [
+        {
+            "image": image,
+            "name": "application-test-sidecar",
+            "ports": [{"containerPort": 8080, "name": "http", "protocol": "TCP"}],
+        }
+    ]
+    assert fn.status.application_image == image
+    assert not fn.spec.image
```

Comment:
This is also duplicated multiple times and in multiple tests - create a function for it